src/app/register-email-form/register-email-form.component.html

```diff
@@ -14,13 +14,16 @@ <h1>{{MESSAGE_PREFIX + '.header'|translate}}</h1>
                     <label class="font-weight-bold"
                            for="email">{{MESSAGE_PREFIX + '.email' | translate}}</label>
                     <input [className]="(email.invalid) && (email.dirty || email.touched) ? 'form-control is-invalid' :'form-control'"
-                           type="text" id="email" formControlName="email"/>
+                           type="text" id="email" formControlName="email"
+                           [attr.aria-label]="MESSAGE_PREFIX + '.aria.label' | translate"
+                           [attr.aria-describedby]="ariaDescribedby"
```

Comment:
@Andrea-Guevara : I think you can make this dynamic in a much easier way. 

Something like this should work:
`[attr.aria-describedby]="(!email.errors) ? '' : (email.errors.required ? 'email-errors-required' : 'email-error-not-valid')"`

That logic is just an if/else structure.  If there are no errors (`!email.errors`) have an empty string in `aria-describedby`.  Otherwise, put in the correct error field ID based on whether `email.errors.required` is true or false.

I think this logic would be a lot easier to understand. It'd also allow you to remove all of your recent changes to the `register-email-form.component.ts` class.

(You'll also see similar if/else structures in other HTML attributes in this same file, so this is more similar to the current code.)

Reply:
@tdonohue Your code suggestion works perfectly. Thank you very much :)